staging/src/k8s.io/apiserver/pkg/audit/policy/checker.go

```diff
@@ -143,14 +143,29 @@ func ruleMatchesResource(r *audit.PolicyRule, attrs authorizer.Attributes) bool
 
 	apiGroup := attrs.GetAPIGroup()
 	resource := attrs.GetResource()
+	resourceWithSubresource := resource
+	if sr := attrs.GetSubresource(); sr != "" {
+		resourceWithSubresource = resourceWithSubresource + "/" + sr
+	}
+
+	name := attrs.GetName()
+
 	for _, gr := range r.Resources {
 		if gr.Group == apiGroup {
 			if len(gr.Resources) == 0 {
 				return true
 			}
-			for _, res := range gr.Resources {
-				if res == resource {
-					return true
+			for _, r := range gr.Resources {
+				// Resource either matches the resource directly or matches the
```

Comment:
I think it's worth to specify which resource is which, e.g. `Resource from the policy rule either matches the target resource directly or target resource + subresource`. WDYT?